contracts/BlendSwap.sol

```diff
@@ -38,6 +38,10 @@ contract BlendSwap {
         blend = IERC20(blend_);
     }
 
+    function getSwap(bytes32 secretHash) public view returns (Swap memory) {
```

Comment:
@kkirka do you have experience with returning structs from _view functions_? Will this work properly?

Reply:
Wait, why do you even need this function? `mapping (bytes32 => Swap) public swaps` is `public`, so Solidity automatically generates the view of the form `function swaps(bytes32) public view returns (Swap memory)`.

Regarding "returning structs from view functions" – yes, it works fine, see, e.g., https://github.com/StakerDAO/blend/blob/master/test/BlendSwap.js#L90